applications/lethe-fluid-matrix-free/fluid_dynamics_matrix_free.cc

```diff
@@ -15,32 +15,81 @@
 
 #include "solvers/fluid_dynamics_matrix_free.h"
 
+#include <core/revision.h>
+
+#include <deal.II/base/revision.h>
+
+std::string
+concatenate_strings(const int argc, char **argv)
+{
+  std::string result = std::string(argv[0]);
+
+  for (int i = 1; i < argc; ++i)
+    result = result + " " + std::string(argv[i]);
+
+  return result;
+}
+
 int
 main(int argc, char *argv[])
 {
   try
     {
       Utilities::MPI::MPI_InitFinalize mpi_initialization(argc, argv, 1);
 
-      if (argc != 2)
+      if (argc == 1)
         {
           std::cout << "Usage:" << argv[0] << " input_file" << std::endl;
           std::exit(1);
         }
 
-      const unsigned int                  dim = get_dimension(argv[1]);
+
+      const std::string file_name(argv[argc - 1]);
+      const bool        print_parameters =
+        (argc == 2) ? false : (std::string(argv[1]) == "--print-parameters");
+
+      const unsigned int                  dim = get_dimension(file_name);
       const Parameters::SizeOfSubsections size_of_subsections =
-        Parameters::get_size_of_subsections(argv[1]);
+        Parameters::get_size_of_subsections(file_name);
+
+      ConditionalOStream pcout(std::cout,
+                               (Utilities::MPI::this_mpi_process(
+                                  MPI_COMM_WORLD) == 0) &&
+                                 print_parameters);
+
+      if (print_parameters)
+        {
+          pcout << "Running: " << concatenate_strings(argc, argv) << std::endl;
+          pcout << "  - deal.II (branch: " << DEAL_II_GIT_BRANCH
+                << "; revision: " << DEAL_II_GIT_REVISION
+                << "; short: " << DEAL_II_GIT_SHORTREV << ")" << std::endl;
+          pcout << "  - Lethe (branch: " << LETHE_GIT_BRANCH
+                << "; revision: " << LETHE_GIT_REVISION
+                << "; short: " << LETHE_GIT_SHORTREV << ")" << std::endl;
```

Comment:
In addition to printing the parameters, I think printing the git version is also useful. These both informations allow to easily reproduce experiments from output files.